File: src-tauri/src/handlers/clinic_info.rs
use crate::datastore::clinic_info;
use crate::types::ClinicInfo;

#[tauri::command]
pub async fn add_clinic_info(data: ClinicInfo, window: tauri::Window) -> Result<String, String> {
    clinic_info::add_clinic_info_db(data, window)
}

#[tauri::command]
pub async fn update_clinic_info(data: ClinicInfo, window: tauri::Window) -> Result<String, String> {
    clinic_info::update_clinic_info_db(data, window)
}

#[tauri::command]
pub async fn get_clinic_info(window: tauri::Window) -> Result<ClinicInfo, String> {
    clinic_info::get_clinic_info_db(window)
}

File: src/api/clinicInfo.ts
import { invoke } from "@tauri-apps/api/core";
import { TClinicInfo } from "../types";

export const getClinicInfoApi = async () => {
  try {
    const res = await invoke<TClinicInfo>("get_clinic_info", {});
    return res;
  } catch (e) {
    console.error("Error getting appointment days:", e);
  }
};
export const addClinicInfoApi = async (data: TClinicInfo) => {
  try {
    const res = await invoke<TClinicInfo>("add_clinic_info", { data });
    return res;
  } catch (e) {
    console.error("Error added clinic info:", e);
  }
};
export const updateClinicInfoApi = async (data: TClinicInfo) => {
  try {
    const res = await invoke<TClinicInfo>("update_clinic_info", { data });
    return res;
  } catch (e) {
    console.error("Error clinic info updat:", e);
  }
};
